lib/src/pub_updater.dart

```diff
@@ -69,12 +72,13 @@ class PubUpdater {
         'activate',
         packageName,
         if (versionConstraint != null) versionConstraint,
+        if (_baseUrl != _defaultBaseUrl) ...['-u', _baseUrl]
```

Comment:
This wont work, it will also need the "source"  to be "hosted"

Reply:
Also, I think it would be better to have params not in the abbr form. In this case, it would be `hosted-uri`.

Reply:
Thinking a little bit more about this, this will be a breaking change since `_baseUrl` so far meant `https://custom-domain.com/api/packages/` not it will mean `https://custom-domain.com`

Reply:
> This wont work, it will also need the "source" to be "hosted"

I think that supporting other types besides hosted doesn't make sense. You would have to write functionality for git to check tags, path doesn't make sense. Since source is set to hosted by default, there is probably no need to override/add to the command parameters.

Reply:
> Thinking a little bit more about this, this will be a breaking change since `_baseUrl` so far meant `https://custom-domain.com/api/packages/` not it will mean `https://custom-domain.com`

This change should not affect any user who uses the default pub.dev server. Users who use self-hosted solutions are not able to activate and update the version because they are missing a parameter in the activate sub-command. I split the address to add the `_baseUrl` parameter to the attributes. this parameter cannot contain the `/api/packages` part, which is the endpoint of pub server and not the baseUrl. But yes, you're right, it could be a breaking change.

example:
```bash
dart pub global activate my_package --hosted-url https://gitea.mydomain.com/api/packages/{owner}/pub/
```
api request:
- base url: https://gitea.mydomain.com/api/packages/{owner}/pub/
- package: my_package
- api endpoint: api/packages/
```
/// final request on self-hosted (gitea, nexus3, etc.)
https://gitea.mydomain.com/api/packages/{owner}/pub/api/packages/my_package/

/// final requuest on pub.dev
https://pub.dev/api/packages/my_package
```
[repository-spec-v2.md](https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md#list-all-versions-of-a-package)